patches/minecraft/net/minecraft/entity/player/PlayerEntity.java.patch

```diff
@@ -440,6 +449,15 @@
        return this.func_208016_c(itextcomponent);
     }
  
+@@ -1878,7 +1997,7 @@
+    }
+ 
+    public EntitySize func_213305_a(Pose p_213305_1_) {
+-      return field_213836_b.getOrDefault(p_213305_1_, field_213835_bs);
++      return field_213836_b.getOrDefault(p_213305_1_, net.minecraftforge.common.PlayerSize.getOrDefault(p_213305_1_,field_213835_bs));
```

Comment:
I can reverse this if everyone agrees.

About making this an instance field: Same reason as above 
a) The instance field has to be properly set every time the player entity is created
b) It would increase patch size, and furthermore include the SIZE_BY_POSE line in the patch, so if Mojang decides to change some size, the patch would have to be updated properly.

Personally, I think it would be better to add yet another (tick) event in this method, or add a modifiable scale factor on per-player basis, which would scale all poses except a few special ones (dead, ...)

Reply:
> a) The instance field has to be properly set every time the player entity is created

Not if it falls back to the existing values. If it is just a map of overrides, we would just check for nonnull and pose in map, otherwise return default.
> b) It would increase patch size, and furthermore include the SIZE_BY_POSE line in the patch, so if Mojang decides to change some size, the patch would have to be updated properly.

I'm not 100% sure how ATs work, but can they be used to remove the static modifier?
Alternatively the `SIZE_BY_POSE` field could be kept as the default, and a `poseSizeOverrides` instance field could be added as the primary lookup instead.